pipeline/usfimr-jrc-iou/main.py

```diff
@@ -0,0 +1,235 @@
+            Prefix=os.path.join(experiment_url.path.lstrip("/"), "predict"),
+        )
+        if list_result["IsTruncated"]:
+            raise ValueError("Didn't get all results, implement ContinuationToken")
+
+        try:
+            s3_keys = list_result["Contents"]
+        except KeyError:
+            print("WARNING: No predictions for {}. Continuing...".format(experiment))
+            continue
+
+        output_csv = os.path.join(
+            args.output_dir, "{}-iou-f1.csv".format(experiment.id)
+        )
+        with open(output_csv, "w") as fp_csv:
+            csv_fieldnames = [
+                "chip_id",
+                "f1_all",
+                "f1_urban",
+                "f1_not_urban",
+                "iou_all",
+                "iou_urban",
+                "iou_not_urban",
+            ]
+            writer = csv.DictWriter(fp_csv, csv_fieldnames)
+            writer.writeheader()
+
+            for obj in s3_keys:
+                chip_id = os.path.basename(obj["Key"])
+
+                prediction_tif_uri = "s3://{}/{}".format(
+                    experiment_url.hostname, obj["Key"]
+                )
+                logger.info("\tprediction: {}".format(prediction_tif_uri))
+
+                # TODO: Encode this in Experiment somehow
+                # Replace chip S1 qualifier with QC to match filenames in ground truth dir
+                if chip_id.startswith("SEN1FLOODS11"):
+                    truth_tif_uri = os.path.join(experiment.ground_truth_dir, chip_id)
+                else:
+                    chip_name = chip_id.replace("_S1.tif", "_QC.tif")
+                    truth_tif_uri = os.path.join(experiment.ground_truth_dir, chip_name)
+                logger.info("\ttruth: {}".format(truth_tif_uri))
+
+                with rasterio.open(prediction_tif_uri) as ds_p, rasterio.open(
+                    truth_tif_uri
+                ) as ds_t, rasterio.open(NLCD_TIF_URI) as ds_nlcd:
+                    nlcd_vrt = WarpedVRT(
+                        ds_nlcd,
+                        crs=ds_p.crs,
+                        height=ds_p.height,
+                        width=ds_p.width,
+                        resampling=Resampling.nearest,
+                        transform=ds_p.transform,
+                        nodata=NODATA,
+                    )
+
+                    p_band = ds_p.read(1).flatten()
+                    t_band = ds_t.read(1).flatten()
+                    nlcd_band = nlcd_vrt.read(1).flatten()
+
+                assert len(p_band) == len(t_band)
+                assert len(p_band) == len(nlcd_band)
+
+                # TODO: Encode this in Experiment somehow...
+                # For two class USFIMR experiments, collapse ground truth three class
+                # labels for water to two class: 2 (flood) + 1 (perm) -> 1 (water)
+                if experiment.id.startswith("USFIMR") and len(experiment.labels) == 1:
+                    t_band = np.where(t_band == 2, 1, t_band)
```

Comment:
This is fine, but I think it might be more idiomatic to just say `t_band = t_band == 2`

Reply:
This was going to blow my mind, but it looks like the result is different for each operation. In this bit of code we want to replace all instances of `2` in the array with `1`:

```
In [1]: import numpy as np
In [2]: a = np.array([0, 1, 1, 2, 2])
In [3]: a == 2
Out[3]: array([False, False, False,  True,  True])
In [4]: np.where(a == 2, 1, a)
Out[4]: array([0, 1, 1, 1, 1])
```